### PR TITLE
Refactoring user bundle

### PR DESCRIPTION
**Pull request by escopecz**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/7995
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Refactoring for Mautic 3

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Test all user and permission forms
3. Including log in and password reset forms

#### List deprecations along with the new alternative:
In https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit#

#### List backwards compatibility breaks:
In https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit#

### CODE REVIEW

File: app/bundles/ApiBundle/Security/Permissions/ApiPermissions.php [hunk 1]
```diff
@@ -47,15 +48,19 @@ public function getName()
...
-            'data'   => (!empty($data['access']) ? $data['access'] : []),
-            'bundle' => 'api',
-            'level'  => 'access',
-        ]);
+        $builder->add(
+            'api:access',
+            PermissionListType::class,
+            [
+                'choices' => [
+                    'full' => 'mautic.api.permissions.granted',
```
Comment (hluchas) [inline app/bundles/ApiBundle/Security/Permissions/ApiPermissions.php]: This should be flipped

Comment (hluchas) [reply app/bundles/ApiBundle/Security/Permissions/ApiPermissions.php]: Please use `choices_as_values` attr

File: app/bundles/CoreBundle/Security/Permissions/AbstractPermissions.php [hunk 2]
```diff
@@ -379,13 +380,17 @@ protected function addStandardFormFields($bundle, $level, &$builder, $data, $inc
...
-            'label'   => $label,
-            'bundle'  => $bundle,
-            'level'   => $level,
-            'data'    => (!empty($data[$level]) ? $data[$level] : []),
-        ]);
+        $builder->add(
+            "$bundle:$level",
+            PermissionListType::class,
+            [
+                'choices' => $choices,
```
Comment (hluchas) [inline app/bundles/CoreBundle/Security/Permissions/AbstractPermissions.php]: The same here

File: app/bundles/ApiBundle/Security/Permissions/ApiPermissions.php (hunk 1 above)
Comment (escopecz) [reply app/bundles/ApiBundle/Security/Permissions/ApiPermissions.php]: Flipped in all places and the choices_as_values added to the class itself. Thanks for noticing!

**Comment by escopecz** [comment]
We decided we won't deal with controllers in M3. It would be very difficult and we plan to rewrite Mautic to API-first solution. It won't generate any HTML. So it would be wasted work.

**Pull request merged**